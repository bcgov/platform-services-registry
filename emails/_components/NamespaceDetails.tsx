import { Heading, Text, Link } from '@react-email/components';
import { defaultProvisionedResourceValues } from '@/constants';

export default function NamespaceDetails({
  cluster,
  licencePlate,
  showNamespaceDetailsTitle = true,
  showDefaultResource = false,
}: {
  cluster: string;
  licencePlate?: string;
  showNamespaceDetailsTitle?: boolean;
  showDefaultResource?: boolean;
}) {
  return (
    <div>
      {showNamespaceDetailsTitle && <Heading className="text-lg">Namespace Details</Heading>}
      <div>
        <Text className="mb-1 font-semibold h-4">OpenShift Cluster: </Text>
        <Text className="mt-1 h-4">{cluster}</Text>
      </div>
      {licencePlate && (
        <>
          <Text className="mb-0 font-semibold h-4">Development Namespace: </Text>
          <Link
            className="mt-1 h-4"
            href={`https://console.apps.${cluster}.devops.gov.bc.ca/k8s/cluster/projects/${licencePlate}-dev`}
          >
            <Text className="mt-1 mb-1 h-4">{licencePlate}-dev</Text>
          </Link>
          {showDefaultResource && (
            <>
              <Text className="mb-1 mt-1 h-4">{defaultProvisionedResourceValues.cpu}</Text>
              <Text className="mb-1 mt-1 h-4">{defaultProvisionedResourceValues.memory}</Text>
              <Text className="mb-1 mt-1 h-4">{defaultProvisionedResourceValues.storage}</Text>
            </>
          )}
          <Text className="mb-0 font-semibold h-4">Test Namespace: </Text>
          <Link
            className="mt-1 h-4"
            href={`https://console.apps.${cluster}.devops.gov.bc.ca/k8s/cluster/projects/${licencePlate}-test`}
          >
            <Text className="mt-1 mb-1 h-4">{licencePlate}-test</Text>
          </Link>
          {showDefaultResource && (
            <>
              <Text className="mb-1 mt-1 h-4">{defaultProvisionedResourceValues.cpu}</Text>
              <Text className="mb-1 mt-1 h-4">{defaultProvisionedResourceValues.memory}</Text>
              <Text className="mb-1 mt-1 h-4">{defaultProvisionedResourceValues.storage}</Text>
            </>
          )}
          <Text className="mb-1 font-semibold h-4">Production Namespace: </Text>
          <Link
            className="mt-1 h-4"
            href={`https://console.apps.${cluster}.devops.gov.bc.ca/k8s/cluster/projects/${licencePlate}-prod`}
          >
            <Text className="mt-1 mb-1 h-4">{licencePlate}-prod</Text>
          </Link>
          {showDefaultResource && (
            <>
              <Text className="mb-1 mt-1 h-4">{defaultProvisionedResourceValues.cpu}</Text>
              <Text className="mb-1 mt-1 h-4">{defaultProvisionedResourceValues.memory}</Text>
              <Text className="mb-1 mt-1 h-4">{defaultProvisionedResourceValues.storage}</Text>
            </>
          )}
          <Text className="mb-1 font-semibold h-4">Tools Namespace: </Text>
          <Link
            className="mt-1 h-4"
            href={`https://console.apps.${cluster}.devops.gov.bc.ca/k8s/cluster/projects/${licencePlate}-tools`}
          >
            <Text className="mt-1 mb-1 h-4">{licencePlate}-tools</Text>
          </Link>
          {showDefaultResource && (
            <>
              <Text className="mb-1 mt-1 h-4">{defaultProvisionedResourceValues.cpu}</Text>
              <Text className="mb-1 mt-1 h-4">{defaultProvisionedResourceValues.memory}</Text>
              <Text className="mb-1 mt-1 h-4">{defaultProvisionedResourceValues.storage}</Text>
            </>
          )}
        </>
      )}
    </div>
  );
}
