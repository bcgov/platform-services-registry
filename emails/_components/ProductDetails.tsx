import { Heading, Link, Text } from '@react-email/components';

interface User {
  firstName: string | null;
  lastName: string | null;
  email: string | null;
  ministry: string | null;
}

export default function ProductDetails({
  name,
  description,
  ministry,
  po,
  tl1,
  tl2,
  licencePlate,
}: {
  name: string;
  description: string;
  ministry: string;
  po: User;
  tl1: User;
  tl2?: User | null;
  licencePlate?: string;
}) {
  return (
    <div>
      <Heading className="text-lg">Product Details</Heading>
      <div>
        <Text className="mb-2 font-semibold h-4">Product Name: </Text>
        <Text className="mt-1 h-4">{name}</Text>
        <Text className="mb-2 font-semibold h-4">Product Description: </Text>
        <Text className="mt-1 h-4">{description}</Text>
        {licencePlate && (
          <div>
            <Text className="mb-2 font-semibold h-4">License Plate: </Text>
            <Text className="mt-0 mb-2 h-4">{licencePlate}</Text>
          </div>
        )}
        <Text className="mb-2 font-semibold h-4">Ministry: </Text>
        <Text className="mt-1 h-4">{ministry}</Text>
        <Text className="mb-2 font-semibold h-4">Product Owner: </Text>
        <Text className="mt-1 mb-2 h-4">
          {po.firstName} {po.lastName}
        </Text>
        <Link className="mt-0 h-4" href={`mailto:${po.email}`}>
          {po.email}
        </Link>
        <Text className="mb-2 font-semibold h-4">Technical Lead: </Text>
        <Text className="mt-0 mb-2 h-4">
          {tl1.firstName} {tl1.lastName}
        </Text>
        <Link className="mt-0 h-4" href={`mailto:${tl1.email}`}>
          {tl1.email}
        </Link>
        {tl2 && (
          <div>
            <Text className="mb-2 font-semibold h-4">Secondary Technical Lead: </Text>
            <Text className="mt-0 mb-2 h-4">
              {tl2.firstName} {tl2.lastName}
            </Text>
            <Link className="mt-0 h-4" href={`mailto:${tl2.email}`}>
              {tl2.email}
            </Link>
          </div>
        )}
      </div>
    </div>
  );
}
