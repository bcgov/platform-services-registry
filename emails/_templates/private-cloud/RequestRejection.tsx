import * as React from 'react';
import { Button, Heading, Text } from '@react-email/components';
import Closing from '@/emails/_components/Closing';
import Comment from '@/emails/_components/Comment';
import QuotaChanges from '@/emails/_components/Edit/QuotaChanges';
import { PrivateCloudRequestWithProjectAndRequestedProject } from '@/request-actions/private-cloud/decision-request';
import { comparePrivateCloudProjects } from '@/emails/_components/Edit/utils/compare-projects';
import Layout from '@/emails/_components/layout/Layout';

interface EmailProp {
  productName: string;
  decisionComment?: string;
  request: PrivateCloudRequestWithProjectAndRequestedProject;
}

const RequestRejectionTemplate = ({ request, productName, decisionComment }: EmailProp) => {
  if (!request || !request.project || !request.decisionData) return <></>;
  const current = request.project;
  const requested = request.decisionData;
  const changed = comparePrivateCloudProjects(current, requested);

  return (
    <Layout>
      <div className="pb-6 mt-4 mb-4 border-solid border-0 border-b-1 border-slate-300">
        <Heading className="text-lg text-black">Sorry, your request was rejected</Heading>
        <Text>Hi Product Team, </Text>
        <Text className="">
          Your request regarding the product {productName} on the Private Cloud Openshift platform has been rejected due
          to the following reason(s):
        </Text>
        <Comment decisionComment={decisionComment} />
        <Text>Log in to the registry and create a new request if the reason(s) above no longer apply.</Text>
        <Button href="https://registry.developer.gov.bc.ca/" className="bg-bcorange rounded-md px-4 py-2 text-white">
          Log in to Registry
        </Button>
      </div>
      <div className="flex flex-row flex-wrap">
        {changed.productionQuota && (
          <QuotaChanges
            licencePlate={`${request.licencePlate}-prod`}
            quotaCurrent={current.productionQuota}
            quotaRequested={requested.productionQuota}
            type="Production"
            cluster={current.cluster}
            currentLabel="Current"
            requestedLabel="Rejected"
          />
        )}
        {changed.testQuota && (
          <QuotaChanges
            licencePlate={`${request.licencePlate}-test`}
            quotaCurrent={current.testQuota}
            quotaRequested={requested.testQuota}
            type="Test"
            cluster={current.cluster}
            currentLabel="Current"
            requestedLabel="Rejected"
          />
        )}
        {changed.developmentQuota && (
          <QuotaChanges
            licencePlate={`${request.licencePlate}-dev`}
            quotaCurrent={current.testQuota}
            quotaRequested={requested.testQuota}
            type="Development"
            cluster={current.cluster}
            currentLabel="Current"
            requestedLabel="Rejected"
          />
        )}
        {changed.toolsQuota && (
          <QuotaChanges
            licencePlate={`${request.licencePlate}-tools`}
            quotaCurrent={current.testQuota}
            quotaRequested={requested.testQuota}
            type="Tools"
            cluster={current.cluster}
            currentLabel="Current"
            requestedLabel="Rejected"
          />
        )}
      </div>

      <div>
        <Closing />
      </div>
    </Layout>
  );
};

export default RequestRejectionTemplate;
