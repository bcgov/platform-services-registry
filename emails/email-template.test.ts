import { render } from '@react-email/render';
import Template, { NewRequest } from './NewRequest';
import Header from './components/Header';

import ProductDetails from './components/ProductDetails';
import Closing from './components/Closing';
import NamespaceDetails from './components/NamespaceDetails';
import { sampleFormData, sampleRequest } from './components/Params';
import RequestRejection from './RequestRejection';
import RequestApproval from './RequestApproval';

describe('Email template snapshots', () => {
  //test components
  it('returns Header component', () => {
    const email = render(Header());
    expect(email).toMatchSnapshot();
  });

  it('returns ProductDetails component', () => {
    const email = render(
      ProductDetails({
        name: sampleFormData.name,
        description: sampleFormData.description,
        ministry: sampleFormData.ministry,
        po: sampleFormData.projectOwner,
        tl1: sampleFormData.primaryTechnicalLead,
        tl2: sampleFormData.secondaryTechnicalLead,
      }),
    );
    expect(email).toMatchSnapshot();
  });

  it('returns NamespaceDetails component', () => {
    const email = render(NamespaceDetails({ cluster: sampleFormData.cluster }));
    expect(email).toMatchSnapshot();
  });

  it('returns Namespace component if approved', () => {
    const email = render(
      NamespaceDetails({
        cluster: sampleRequest.requestedProject.cluster,
        licencePlate: sampleRequest.licencePlate,
      }),
    );
    expect(email).toMatchSnapshot();
  });

  it('returns Closing component', () => {
    const email = render(Closing());
    expect(email).toMatchSnapshot();
  });

  //test templates
  it('returns NewRequest template', () => {
    const formData = sampleFormData;
    const email = render(NewRequest({ formData: sampleFormData }), { pretty: true });
    expect(email).toMatchSnapshot();
  });

  it('returns NewRequest template', () => {
    const email = render(RequestApproval({ request: sampleRequest }), { pretty: true });
    expect(email).toMatchSnapshot();
  });

  it('returns NewRequest template', () => {
    const email = render(RequestRejection({ request: sampleRequest }), { pretty: true });
    expect(email).toMatchSnapshot();
  });
});
