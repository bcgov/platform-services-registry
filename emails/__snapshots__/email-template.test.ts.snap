// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Email template snapshots returns Closing component 1`] = `"<!DOCTYPE html PUBLIC "-//W3C//DTD XHTML 1.0 Transitional//EN" "http://www.w3.org/TR/xhtml1/DTD/xhtml1-transitional.dtd"><div><p style="font-size:14px;line-height:24px;margin:16px 0">If you have any questions, send us a message at <a class="mt-0 h-4" href="mailto:PlatformServicesTeam@gov.bc.ca" style="color:#067df7;text-decoration:none" target="_blank">PlatformServicesTeam@gov.bc.ca</a>. We&#x27;d love to hear from you.</p><p style="font-size:14px;line-height:24px;margin:16px 0">-- The Registry Team</p></div>"`;

exports[`Email template snapshots returns Header component 1`] = `"<!DOCTYPE html PUBLIC "-//W3C//DTD XHTML 1.0 Transitional//EN" "http://www.w3.org/TR/xhtml1/DTD/xhtml1-transitional.dtd"><div class="flex flex-row border-b-3 border-bcorange bg-bcblue shadow"><img class="m-auto ml-2 mr-2" alt="BC Platform Services Product Registry" height="41" src="https://dev-pltsvc.apps.silver.devops.gov.bc.ca/private-cloud/products/logo.png" style="display:block;outline:none;border:none;text-decoration:none" width="58"/><div class="flex flex-row text-white"><p class="text-lg mr-2 font-roboto font-thin" style="font-size:14px;line-height:24px;margin:16px 0">BC Platform Services</p><p class="text-lg font-roboto font-normal" style="font-size:14px;line-height:24px;margin:16px 0">Product Registry</p></div></div>"`;

exports[`Email template snapshots returns Namespace component if approved 1`] = `"<!DOCTYPE html PUBLIC "-//W3C//DTD XHTML 1.0 Transitional//EN" "http://www.w3.org/TR/xhtml1/DTD/xhtml1-transitional.dtd"><div><h1 class="text-lg">Namespace Details</h1><div><p class="mb-0 font-semibold h-4" style="font-size:14px;line-height:24px;margin:16px 0">OpenShift Cluster: </p><p class="mt-0 h-4" style="font-size:14px;line-height:24px;margin:16px 0">SILVER</p></div><p class="mb-0 font-semibold h-4" style="font-size:14px;line-height:24px;margin:16px 0">Development Namespace: </p><p class="mt-0 h-4" style="font-size:14px;line-height:24px;margin:16px 0">be74f8-dev</p><p class="mb-0 font-semibold h-4" style="font-size:14px;line-height:24px;margin:16px 0">Test Namespace: </p><p class="mt-0 h-4" style="font-size:14px;line-height:24px;margin:16px 0">be74f8-test</p><p class="mb-0 font-semibold h-4" style="font-size:14px;line-height:24px;margin:16px 0">Production Namespace: </p><p class="mt-0 h-4" style="font-size:14px;line-height:24px;margin:16px 0">be74f8-prod</p><p class="mb-0 font-semibold h-4" style="font-size:14px;line-height:24px;margin:16px 0">Tools Namespace: </p><p class="mt-0 h-4" style="font-size:14px;line-height:24px;margin:16px 0">be74f8-tools</p></div>"`;

exports[`Email template snapshots returns NamespaceDetails component 1`] = `"<!DOCTYPE html PUBLIC "-//W3C//DTD XHTML 1.0 Transitional//EN" "http://www.w3.org/TR/xhtml1/DTD/xhtml1-transitional.dtd"><div><h1 class="text-lg">Namespace Details</h1><div><p class="mb-0 font-semibold h-4" style="font-size:14px;line-height:24px;margin:16px 0">OpenShift Cluster: </p><p class="mt-0 h-4" style="font-size:14px;line-height:24px;margin:16px 0">KLAB</p></div></div>"`;

exports[`Email template snapshots returns NewRequest template 1`] = `
"<!DOCTYPE html PUBLIC "-//W3C//DTD XHTML 1.0 Transitional//EN" "http://www.w3.org/TR/xhtml1/DTD/xhtml1-transitional.dtd">
<html dir="ltr" lang="en">
  <div style="border-width:1px;border-style:solid;border-color:rgb(234,234,234);border-radius:0.25rem;margin-top:1rem;margin-bottom:1rem;margin-left:auto;margin-right:auto;padding:1rem;max-width:36rem">
    <div class="flex flex-row border-b-3 border-bcorange bg-bcblue shadow"><img class="m-auto ml-2 mr-2" alt="BC Platform Services Product Registry" height="41" src="https://dev-pltsvc.apps.silver.devops.gov.bc.ca/private-cloud/products/logo.png" style="display:block;outline:none;border:none;text-decoration:none" width="58" />
      <div class="flex flex-row text-white">
        <p class="text-lg mr-2 font-roboto font-thin" style="font-size:14px;line-height:24px;margin:16px 0">BC Platform Services</p>
        <p class="text-lg font-roboto font-normal" style="font-size:14px;line-height:24px;margin:16px 0">Product Registry</p>
      </div>
    </div>

    <body style="background-color:rgb(255,255,255);margin-top:auto;margin-bottom:auto;margin-left:auto;margin-right:auto;font-family:Inter var, ui-sans-serif, system-ui, -apple-system, BlinkMacSystemFont, &quot;Segoe UI&quot;, Roboto, &quot;Helvetica Neue&quot;, Arial, &quot;Noto Sans&quot;, sans-serif, &quot;Apple Color Emoji&quot;, &quot;Segoe UI Emoji&quot;, &quot;Segoe UI Symbol&quot;, &quot;Noto Color Emoji&quot;;font-size:0.75rem;line-height:1rem">
      <div style="margin:3rem">
        <div style="padding-bottom:1.5rem;margin-top:1rem;margin-bottom:1rem;border-style:solid;border-width:0px;border-bottom-width:1px;border-color:rgb(203,213,225)">
          <h1 style="font-size:1.125rem;line-height:1.75rem">New Provisioning Request!</h1>
          <p style="font-size:14px;line-height:24px;margin:16px 0">Hi Registry Team, </p>
          <p class="" style="font-size:14px;line-height:24px;margin:16px 0">There is a new request that requires your review. Log in to the Registry to review the details. If you have any questions about the request, the PO and TL contact details are included below and in the Registry</p><a href="https://dev-pltsvc.apps.silver.devops.gov.bc.ca/private-cloud/products" style="background-color:rgb(252,186,25);border-radius:0.375rem;padding-left:1rem;padding-right:1rem;padding-top:0.5rem;padding-bottom:0.5rem;color:rgb(255,255,255);line-height:100%;text-decoration:none;display:inline-block;max-width:100%;padding:8px 16px 8px 16px" target="_blank"><span><!--[if mso]><i style="letter-spacing: 16px;mso-font-width:-100%;mso-text-raise:12" hidden>&nbsp;</i><![endif]--></span><span style="max-width:100%;display:inline-block;line-height:120%;mso-padding-alt:0px;mso-text-raise:6px">Review Request</span><span><!--[if mso]><i style="letter-spacing: 16px;mso-font-width:-100%" hidden>&nbsp;</i><![endif]--></span></a>
        </div>
        <div style="padding-bottom:1.5rem;margin-top:1rem;margin-bottom:1rem;border-style:solid;border-width:0px;border-bottom-width:1px;border-color:rgb(203,213,225)">
          <div>
            <h1 class="text-lg">Product Details</h1>
            <div>
              <p class="mb-0 font-semibold h-4" style="font-size:14px;line-height:24px;margin:16px 0">Product Name: </p>
              <p class="mt-0 h-4" style="font-size:14px;line-height:24px;margin:16px 0">placeholder</p>
              <p class="mb-0 font-semibold h-4" style="font-size:14px;line-height:24px;margin:16px 0">Product Description: </p>
              <p class="mt-0 h-4" style="font-size:14px;line-height:24px;margin:16px 0">placeholder</p>
              <p class="mb-0 font-semibold h-4" style="font-size:14px;line-height:24px;margin:16px 0">Ministry: </p>
              <p class="mt-0 h-4" style="font-size:14px;line-height:24px;margin:16px 0">CITZ</p>
              <p class="mb-0 font-semibold h-4" style="font-size:14px;line-height:24px;margin:16px 0">Product Owner: </p>
              <p class="mt-0 mb-0 h-4" style="font-size:14px;line-height:24px;margin:16px 0">John Doe</p><a class="mt-0 h-4" href="mailto:john.doe@gov.bc.ca" style="color:#067df7;text-decoration:none" target="_blank">john.doe@gov.bc.ca</a>
              <p class="mb-0 font-semibold h-4" style="font-size:14px;line-height:24px;margin:16px 0">Product Owner: </p>
              <p class="mt-0 mb-0 h-4" style="font-size:14px;line-height:24px;margin:16px 0">Sarah Williams</p><a class="mt-0 h-4" href="mailto:sarah.williams@gov.bc.ca" style="color:#067df7;text-decoration:none" target="_blank">sarah.williams@gov.bc.ca</a>
            </div>
          </div>
        </div>
        <div style="padding-bottom:1.5rem;margin-top:1rem;margin-bottom:1rem;border-style:solid;border-width:0px;border-bottom-width:1px;border-color:rgb(203,213,225)">
          <div>
            <h1 class="text-lg">Namespace Details</h1>
            <div>
              <p class="mb-0 font-semibold h-4" style="font-size:14px;line-height:24px;margin:16px 0">OpenShift Cluster: </p>
              <p class="mt-0 h-4" style="font-size:14px;line-height:24px;margin:16px 0">KLAB</p>
            </div>
          </div>
        </div>
        <div>
          <div>
            <p style="font-size:14px;line-height:24px;margin:16px 0">If you have any questions, send us a message at <a class="mt-0 h-4" href="mailto:PlatformServicesTeam@gov.bc.ca" style="color:#067df7;text-decoration:none" target="_blank">PlatformServicesTeam@gov.bc.ca</a>. We&#x27;d love to hear from you.</p>
            <p style="font-size:14px;line-height:24px;margin:16px 0">-- The Registry Team</p>
          </div>
        </div>
      </div>
    </body>
  </div>

</html>"
`;

exports[`Email template snapshots returns NewRequest template 2`] = `
"<!DOCTYPE html PUBLIC "-//W3C//DTD XHTML 1.0 Transitional//EN" "http://www.w3.org/TR/xhtml1/DTD/xhtml1-transitional.dtd">
<html dir="ltr" lang="en">
  <div style="border-width:1px;border-style:solid;border-color:rgb(234,234,234);border-radius:0.25rem;margin-top:1rem;margin-bottom:1rem;margin-left:auto;margin-right:auto;padding:1rem;max-width:36rem">
    <div class="flex flex-row border-b-3 border-bcorange bg-bcblue shadow"><img class="m-auto ml-2 mr-2" alt="BC Platform Services Product Registry" height="41" src="https://dev-pltsvc.apps.silver.devops.gov.bc.ca/private-cloud/products/logo.png" style="display:block;outline:none;border:none;text-decoration:none" width="58" />
      <div class="flex flex-row text-white">
        <p class="text-lg mr-2 font-roboto font-thin" style="font-size:14px;line-height:24px;margin:16px 0">BC Platform Services</p>
        <p class="text-lg font-roboto font-normal" style="font-size:14px;line-height:24px;margin:16px 0">Product Registry</p>
      </div>
    </div>

    <body style="background-color:rgb(255,255,255);margin-top:auto;margin-bottom:auto;margin-left:auto;margin-right:auto;font-family:Inter var, ui-sans-serif, system-ui, -apple-system, BlinkMacSystemFont, &quot;Segoe UI&quot;, Roboto, &quot;Helvetica Neue&quot;, Arial, &quot;Noto Sans&quot;, sans-serif, &quot;Apple Color Emoji&quot;, &quot;Segoe UI Emoji&quot;, &quot;Segoe UI Symbol&quot;, &quot;Noto Color Emoji&quot;;font-size:0.75rem;line-height:1rem">
      <div style="margin:3rem">
        <div style="padding-bottom:1.5rem;margin-top:1rem;margin-bottom:1rem;border-style:solid;border-width:0px;border-bottom-width:1px;border-color:rgb(203,213,225)">
          <h1 style="font-size:1.125rem;line-height:1.75rem">Hurray! Your provisioning request was approved and completed!</h1>
          <p style="font-size:14px;line-height:24px;margin:16px 0">Hi John, </p>
          <p class="" style="font-size:14px;line-height:24px;margin:16px 0">Your request for a new project set for your product on the Private Cloud Openshift platform is complete. Log in to the cluster console using the button bellow and you&#x27;ll see all four namespaces included in a project set. If you have any more questions, reach out to the Platform Services team in the RocketChat channel #devops-operations.</p>
          <p class="" style="font-size:14px;line-height:24px;margin:16px 0">The Product Owner and the Technical Lead have been provisioned with admin access to the namespaces above and can add other users as necessary. Please note that if a Product Owner or a Technical Lead is removed as a project contact in the Platform Registry, they will lost their access to the projcect set namespaces in Openshift. The new Product or Technical Lead provided on the product details page will gain the administrative access to the namespaces.</p><a href="https://dev-pltsvc.apps.silver.devops.gov.bc.ca/private-cloud/products" style="background-color:rgb(252,186,25);border-radius:0.375rem;padding-left:1rem;padding-right:1rem;padding-top:0.5rem;padding-bottom:0.5rem;color:rgb(255,255,255);line-height:100%;text-decoration:none;display:inline-block;max-width:100%;padding:8px 16px 8px 16px" target="_blank"><span><!--[if mso]><i style="letter-spacing: 16px;mso-font-width:-100%;mso-text-raise:12" hidden>&nbsp;</i><![endif]--></span><span style="max-width:100%;display:inline-block;line-height:120%;mso-padding-alt:0px;mso-text-raise:6px">Log in to Console</span><span><!--[if mso]><i style="letter-spacing: 16px;mso-font-width:-100%" hidden>&nbsp;</i><![endif]--></span></a>
        </div>
        <div style="padding-bottom:1.5rem;margin-top:1rem;margin-bottom:1rem;border-style:solid;border-width:0px;border-bottom-width:1px;border-color:rgb(203,213,225)">
          <div>
            <h1 class="text-lg">Product Details</h1>
            <div>
              <p class="mb-0 font-semibold h-4" style="font-size:14px;line-height:24px;margin:16px 0">Product Name: </p>
              <p class="mt-0 h-4" style="font-size:14px;line-height:24px;margin:16px 0">31.1</p>
              <p class="mb-0 font-semibold h-4" style="font-size:14px;line-height:24px;margin:16px 0">Product Description: </p>
              <p class="mt-0 h-4" style="font-size:14px;line-height:24px;margin:16px 0">1</p>
              <p class="mb-0 font-semibold h-4" style="font-size:14px;line-height:24px;margin:16px 0">Ministry: </p>
              <p class="mt-0 h-4" style="font-size:14px;line-height:24px;margin:16px 0">CITZ</p>
              <p class="mb-0 font-semibold h-4" style="font-size:14px;line-height:24px;margin:16px 0">Product Owner: </p>
              <p class="mt-0 mb-0 h-4" style="font-size:14px;line-height:24px;margin:16px 0">John Doe</p><a class="mt-0 h-4" href="mailto:john.doe@gov.bc.ca" style="color:#067df7;text-decoration:none" target="_blank">john.doe@gov.bc.ca</a>
              <p class="mb-0 font-semibold h-4" style="font-size:14px;line-height:24px;margin:16px 0">Product Owner: </p>
              <p class="mt-0 mb-0 h-4" style="font-size:14px;line-height:24px;margin:16px 0">Sarah Williams</p><a class="mt-0 h-4" href="mailto:sarah.williams@gov.bc.ca" style="color:#067df7;text-decoration:none" target="_blank">sarah.williams@gov.bc.ca</a>
            </div>
          </div>
        </div>
        <div style="padding-bottom:1.5rem;margin-top:1rem;margin-bottom:1rem;border-style:solid;border-width:0px;border-bottom-width:1px;border-color:rgb(203,213,225)">
          <div>
            <h1 class="text-lg">Namespace Details</h1>
            <div>
              <p class="mb-0 font-semibold h-4" style="font-size:14px;line-height:24px;margin:16px 0">OpenShift Cluster: </p>
              <p class="mt-0 h-4" style="font-size:14px;line-height:24px;margin:16px 0">SILVER</p>
            </div>
            <p class="mb-0 font-semibold h-4" style="font-size:14px;line-height:24px;margin:16px 0">Development Namespace: </p>
            <p class="mt-0 h-4" style="font-size:14px;line-height:24px;margin:16px 0">be74f8-dev</p>
            <p class="mb-0 font-semibold h-4" style="font-size:14px;line-height:24px;margin:16px 0">Test Namespace: </p>
            <p class="mt-0 h-4" style="font-size:14px;line-height:24px;margin:16px 0">be74f8-test</p>
            <p class="mb-0 font-semibold h-4" style="font-size:14px;line-height:24px;margin:16px 0">Production Namespace: </p>
            <p class="mt-0 h-4" style="font-size:14px;line-height:24px;margin:16px 0">be74f8-prod</p>
            <p class="mb-0 font-semibold h-4" style="font-size:14px;line-height:24px;margin:16px 0">Tools Namespace: </p>
            <p class="mt-0 h-4" style="font-size:14px;line-height:24px;margin:16px 0">be74f8-tools</p>
          </div>
        </div>
        <div>
          <div>
            <p style="font-size:14px;line-height:24px;margin:16px 0">If you have any questions, send us a message at <a class="mt-0 h-4" href="mailto:PlatformServicesTeam@gov.bc.ca" style="color:#067df7;text-decoration:none" target="_blank">PlatformServicesTeam@gov.bc.ca</a>. We&#x27;d love to hear from you.</p>
            <p style="font-size:14px;line-height:24px;margin:16px 0">-- The Registry Team</p>
          </div>
        </div>
      </div>
    </body>
  </div>

</html>"
`;

exports[`Email template snapshots returns NewRequest template 3`] = `
"<!DOCTYPE html PUBLIC "-//W3C//DTD XHTML 1.0 Transitional//EN" "http://www.w3.org/TR/xhtml1/DTD/xhtml1-transitional.dtd">
<html dir="ltr" lang="en">
  <div style="border-width:1px;border-style:solid;border-color:rgb(234,234,234);border-radius:0.25rem;margin-top:1rem;margin-bottom:1rem;margin-left:auto;margin-right:auto;padding:1rem;max-width:36rem">
    <div class="flex flex-row border-b-3 border-bcorange bg-bcblue shadow"><img class="m-auto ml-2 mr-2" alt="BC Platform Services Product Registry" height="41" src="https://dev-pltsvc.apps.silver.devops.gov.bc.ca/private-cloud/products/logo.png" style="display:block;outline:none;border:none;text-decoration:none" width="58" />
      <div class="flex flex-row text-white">
        <p class="text-lg mr-2 font-roboto font-thin" style="font-size:14px;line-height:24px;margin:16px 0">BC Platform Services</p>
        <p class="text-lg font-roboto font-normal" style="font-size:14px;line-height:24px;margin:16px 0">Product Registry</p>
      </div>
    </div>

    <body style="background-color:rgb(255,255,255);margin-top:auto;margin-bottom:auto;margin-left:auto;margin-right:auto;font-family:Inter var, ui-sans-serif, system-ui, -apple-system, BlinkMacSystemFont, &quot;Segoe UI&quot;, Roboto, &quot;Helvetica Neue&quot;, Arial, &quot;Noto Sans&quot;, sans-serif, &quot;Apple Color Emoji&quot;, &quot;Segoe UI Emoji&quot;, &quot;Segoe UI Symbol&quot;, &quot;Noto Color Emoji&quot;;font-size:0.75rem;line-height:1rem">
      <div style="margin:3rem">
        <div style="padding-bottom:1.5rem;margin-top:1rem;margin-bottom:1rem;border-style:solid;border-width:0px;border-bottom-width:1px;border-color:rgb(203,213,225)">
          <h1 style="font-size:1.125rem;line-height:1.75rem">Sorry, your provisioning request was Rejected</h1>
          <p style="font-size:14px;line-height:24px;margin:16px 0">Hi John, </p>
          <p class="" style="font-size:14px;line-height:24px;margin:16px 0">Your request for a new project set for your product on the Private Cloud Openshift platform is rejected due to the following reason(s):</p>
          <p class="" style="font-size:14px;line-height:24px;margin:16px 0">sample reason</p>
          <p style="font-size:14px;line-height:24px;margin:16px 0">Log in to your registry account and raise a new request if the above rejection reason no longer applies</p><a href="https://dev-pltsvc.apps.silver.devops.gov.bc.ca/private-cloud/products" style="background-color:rgb(252,186,25);border-radius:0.375rem;padding-left:1rem;padding-right:1rem;padding-top:0.5rem;padding-bottom:0.5rem;color:rgb(255,255,255);line-height:100%;text-decoration:none;display:inline-block;max-width:100%;padding:8px 16px 8px 16px" target="_blank"><span><!--[if mso]><i style="letter-spacing: 16px;mso-font-width:-100%;mso-text-raise:12" hidden>&nbsp;</i><![endif]--></span><span style="max-width:100%;display:inline-block;line-height:120%;mso-padding-alt:0px;mso-text-raise:6px">Log in to Registry</span><span><!--[if mso]><i style="letter-spacing: 16px;mso-font-width:-100%" hidden>&nbsp;</i><![endif]--></span></a>
        </div>
        <div>
          <div>
            <p style="font-size:14px;line-height:24px;margin:16px 0">If you have any questions, send us a message at <a class="mt-0 h-4" href="mailto:PlatformServicesTeam@gov.bc.ca" style="color:#067df7;text-decoration:none" target="_blank">PlatformServicesTeam@gov.bc.ca</a>. We&#x27;d love to hear from you.</p>
            <p style="font-size:14px;line-height:24px;margin:16px 0">-- The Registry Team</p>
          </div>
        </div>
      </div>
    </body>
  </div>

</html>"
`;

exports[`Email template snapshots returns ProductDetails component 1`] = `"<!DOCTYPE html PUBLIC "-//W3C//DTD XHTML 1.0 Transitional//EN" "http://www.w3.org/TR/xhtml1/DTD/xhtml1-transitional.dtd"><div><h1 class="text-lg">Product Details</h1><div><p class="mb-0 font-semibold h-4" style="font-size:14px;line-height:24px;margin:16px 0">Product Name: </p><p class="mt-0 h-4" style="font-size:14px;line-height:24px;margin:16px 0">placeholder</p><p class="mb-0 font-semibold h-4" style="font-size:14px;line-height:24px;margin:16px 0">Product Description: </p><p class="mt-0 h-4" style="font-size:14px;line-height:24px;margin:16px 0">placeholder</p><p class="mb-0 font-semibold h-4" style="font-size:14px;line-height:24px;margin:16px 0">Ministry: </p><p class="mt-0 h-4" style="font-size:14px;line-height:24px;margin:16px 0">CITZ</p><p class="mb-0 font-semibold h-4" style="font-size:14px;line-height:24px;margin:16px 0">Product Owner: </p><p class="mt-0 mb-0 h-4" style="font-size:14px;line-height:24px;margin:16px 0">John Doe</p><a class="mt-0 h-4" href="mailto:john.doe@gov.bc.ca" style="color:#067df7;text-decoration:none" target="_blank">john.doe@gov.bc.ca</a><p class="mb-0 font-semibold h-4" style="font-size:14px;line-height:24px;margin:16px 0">Product Owner: </p><p class="mt-0 mb-0 h-4" style="font-size:14px;line-height:24px;margin:16px 0">Sarah Williams</p><a class="mt-0 h-4" href="mailto:sarah.williams@gov.bc.ca" style="color:#067df7;text-decoration:none" target="_blank">sarah.williams@gov.bc.ca</a></div></div>"`;
